File: matlab/read_images.py
import os
import numpy as np

# Get current path
current_path = os.getcwd()
print(f"The current path is: {current_path}")

# Define parameters
images = range(19, 890)  # This creates a range equivalent to 19:889 in MATLAB
datatype = np.uint16  # The numpy equivalent for MATLAB's 'uint16'
im_size = (400, 400)  # In python, tuples are often used to represent sizes or shapes
endianness = '<'  # Little-endian. Equivalent to 'b' in MATLAB.
filename = "../data/No_metal/acquisition_0/IE1705794_P406.i18{:04d}.raw"

sino = np.zeros((400, 400, 871))

# Loop through and read the images
for i in images:
    im_name = filename.format(i)
    with open(im_name, 'rb') as f:
        # Read the data
        V = np.fromfile(f, dtype=datatype)

        # Reshape it
        sino[:, :, i - 18] = V.reshape(im_size)

# Get background intensity
intensity = np.mean(sino[-21:, 100:300, 349])

sino = -np.log(sino / intensity)
sino[sino < 0] = 0

# Save the data
np.save("sino.npy", sino)  # This saves the data in numpy's .npy format

File: matlab/read_images.m
currentPath = pwd;

disp(['The current path is: ', currentPath]);



images = 0:871;
datatype = 'uint16';
im_size = [400,400];
endianness = 'b';
filename = "../data/No_metal/acquisition_0/IE1705794_P406.i18%04d.raw";

sino = zeros(400,400,871);

for i = images
    im_name = sprintf(filename,i);

    fid = fopen(im_name, 'r');

    % read the data
    V = fread(fid, inf, ['*' datatype], 0, endianness);
    fclose(fid);

    % shape it
    sino(:,:,i+1) = reshape(V, im_size);
end

% Get background intensity
intensity = mean(sino(end-20:end,100:300,350),'all');

sino = -log(sino/intensity);
sino(sino < 0) = 0;

save("sino","sino","-v7.3")
